mne/epochs.py

```diff
@@ -3249,6 +3255,42 @@ def __init__(
         # keep track of original sfreq (needed for annotations)
         raw_sfreq = raw.info["sfreq"]
 
+        # get events from annotations if no events given
+        if events is None:
+            events, event_id_tmp = events_from_annotations(raw)
+            if events.size == 0:
+                raise RuntimeError(
+                    "No annotations found in the raw object."
```

Comment:
```suggestion
                    "No usable annotations found in the raw object. "
```
we automatically ignore BAD and EDGE annots